src/Features/Customer/UserProfile/Profile.js

```diff
@@ -1,162 +1,148 @@
 import React, { useState } from 'react';
 
+import { useMutation } from 'react-query';
+import { useSelector, useDispatch } from 'react-redux';
+
 import CommonGridBasedForm from '../../../components/CommonGridBasedForm';
 import { TEXT_FIELD } from '../../../components/CommonGridBasedForm/FieldTypes';
 import { emailRegex } from '../../../redux/ActionTypes';
 import { contactRegex } from '../../../scripts/constants';
+import { fieldChangeHandler, validateOnSubmit } from '../../../util/CommonGridBasedFormUtils';
+import { updateUserData } from '../../Auth/actions';
+import { editUser } from './mutation';
 
 const AddUser = () => {
-  const [onSaveSuccess, setOnSaveSuccess] = useState(false);
-  const [isEdit] = useState(true);
-  const validateOnSubmit = () => {
-    let isValid = true;
-
-    const ValidateArray = fields.map((field) => {
-      if (
-        field.value === '' ||
-        field.value === undefined ||
-        field.value === null ||
-        (field.value.constructor.name == 'Array' && field.value.length === 0)
-      ) {
-        isValid = false;
-
-        field.errorMessage = field.label + ' field cannot be empty';
-
-        field.isValid = false;
-
-        return field;
-      }
-
-      field.isValid = true;
-
-      field.errorMessage = '';
-
-      !isValid ? null : (isValid = field.isValid);
-
-      return field;
-    });
-
-    setFields(ValidateArray);
-
-    return isValid;
-  };
-
-  const [, setEmail] = useState('');
-
-  const [, setContact] = useState('');
-
-  const [, setName] = useState('');
+  const { id, email, name, token } = useSelector((state) => {
+    const {
+      authReducer: {
+        id,
+        email,
+        name,
+        accessToken: { token },
+      },
+    } = state;
+    return {
+      id,
+      email,
+      name,
+      token,
+    };
+  });
+
+  const dispatch = useDispatch();
+
+  const { isLoading, mutate: editUserMutate } = useMutation(editUser, {
+    onSuccess: async (resData) => {
+      if (!resData.code) {
```

Comment:
I think either it is error or success the API will always return status code so whats the reason to check the code here ??

Reply:
on success it does not, only in error